client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -193,37 +194,35 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
 
     batchHandleChangePartitionSchedulerThread.foreach {
       _.scheduleAtFixedRate(
-        new Runnable {
-          override def run(): Unit = {
-            try {
-              changePartitionRequests.asScala.foreach { case (shuffleId, requests) =>
-                requests.synchronized {
-                  batchHandleChangePartitionExecutors.submit {
-                    new Runnable {
-                      override def run(): Unit = {
-                        // For each partition only need handle one request
-                        val distinctPartitions = requests.asScala.filter { case (partitionId, _) =>
-                          !inBatchPartitions.get(shuffleId).contains(partitionId)
-                        }.map { case (partitionId, request) =>
-                          inBatchPartitions.get(shuffleId).add(partitionId)
-                          request.asScala.toArray.maxBy(_.epoch)
-                        }.toArray
-                        if (distinctPartitions.nonEmpty) {
-                          batchHandleChangePartitions(
-                            distinctPartitions.head.applicationId,
-                            shuffleId,
-                            distinctPartitions)
-                        }
+        () => {
```

Comment:
this could not be simplified because the project still supports Spark2.4 w/ Scala 2.11, which does not support SAM (in default)

Reply:
> this could not be simplified because the project still supports Spark 2.4 w/ Scala 2.11, which does not support SAM (in default)

done